### DISCUSSION TITLE
How can I make the future events to show?

### DISCUSSION THREAD

**Discussion by azegas** [MMRIZE/MMM-CalendarExt3#27] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/78803192/187453623-9f2c49e8-01b8-446e-8713-0155f0e19abd.png)

I have some events on September 6th, 8th, 9th.. how can I make them visible? I can not find such function in the documentation.

Thank you!

**Comment by eouia** [comment]
I need your configuration of `calendar` and `MMM-CalendarExt3` modules. Could you send me the `config.js` file through mail? (eouia0819@gmail.com)

**Comment by eouia** [comment]
Ok. It is probably already a known issue. 
It seems related to the default calendar module's behaviours - `maximumEntries (default 10)`, `maximumNumberOfDays (default 365)`. 
When you set `broadcastPastEvents` to `true`, the calendar module will emit the past events, but from the older events by `maximumEntries` and `maximumNumberOfDays`. That's why your screenshot shows only `10` old events.

I recommend `maximumNumberOfDays: 45` and `maximumEntries: 100` (adjust by density of your events. If you have almost 100 events per month, set it enough.)

**Comment by eouia** [comment, marked as answer]
1. broadcastPastEvents: false, maximumEntries: 10 
![image](https://user-images.githubusercontent.com/1720610/187496666-fa2b1221-ae56-4fab-b958-3afb9957e85f.png)
Only future events will be shown.


2. broadcastPastEvents: true, maximumEntries: 10
![image](https://user-images.githubusercontent.com/1720610/187496811-35b32d7c-8dc2-4f50-a07b-4ddf984771e5.png)
Nothing is shown because old events would consume all ten quotas.


3. broadcastPastEvents: true, maximumEntries: 100
![image](https://user-images.githubusercontent.com/1720610/187497083-2d934cd6-3e14-415e-b2bd-b52a518ca70e.png)
Now some old events are displayed as usual because quotas are enough (maximumEntries: 100)

4. broadcastPastEvents: true, maximumEntries: 10, maximumNumberOfDays: 14
![image](https://user-images.githubusercontent.com/1720610/187497155-f43bbf4c-fed5-4de7-8d2f-71e0cd054121.png)
Events over 14 days ago are not showing.

**Reply by azegas** [reply]
Thank you, this was the solution. 

Also I had to make sure to write 

broadcastPastEvents: true, 
maximumEntries: 10, m
aximumNumberOfDays: 14 

in "calendar" module and not in "MMM-CalendarExt3" module.

**Reply by eouia** [reply]
If you have many calendars and events, you might need to raise `maximumEntries`. Of course, you can find the optimal value by shoot-and-aim. :)
